src/main/java/seedu/address/logic/commands/ViewCommand.java

```diff
@@ -0,0 +1,140 @@
+import static seedu.address.logic.parser.ViewCommandParser.CLAIM_BUDGET;
+import static seedu.address.logic.parser.ViewCommandParser.DEPARTMENT;
+import static seedu.address.logic.parser.ViewCommandParser.EMAIL_IDENTIFIER;
+import static seedu.address.logic.parser.ViewCommandParser.NAME_IDENTIFIER;
+import static seedu.address.logic.parser.ViewCommandParser.PHONE_IDENTIFIER;
+import static seedu.address.logic.parser.ViewCommandParser.SALARY_IDENTIFIER;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Objects;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+
+/**
+ * Returns ViewCommand Object based on what user's wants to view.
+ */
+public class ViewCommand extends Command {
+
+    public static final String COMMAND_WORD = "view";
+    public static final String WRONG_PREFIX = "Allowed Formats: n/, a/, e/, s/, b/, d/, dob/.\n"
+            + "Example: view n/1,2 e/1,2.";
+    public final HashMap<String, List<Index>> references;
+
+    /**
+     * Initialises ViewCommand Object containing view operations on specific individuals.
+     *
+     * @param references HashMap object referencing view operations on Index individuals in list.
+     */
+    public ViewCommand(HashMap<String, List<Index>> references) {
+        this.references = references;
+    }
+
+    /**
+     * Returns CommandResult after successfully parsing user's view commands.
+     *
+     * @param model {@code Model} which the command should operate on.
+     * @return CommandResult which contains string representation of person's attribute.
+     * @throws CommandException Exception thrown if index input from HR is beyond the pre-existing max list index.
+     */
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        String finalString = "";
+
+        for (String key: references.keySet()) {
+            List<Index> indexList = references.get(key);
+            String tempStr = "You are viewing " + key + ":\n";
+            for (int i = 0; i < indexList.size(); i++) {
+                int listIndex = indexList.get(i).getZeroBased();
+                if (listIndex >= lastShownList.size()) {
+                    throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+                }
+                Person personToEdit = lastShownList.get(indexList.get(i).getZeroBased());
+                HashMap<String, String> personRelevantInfo = generateArrInfo(personToEdit);
+                int index = indexList.get(i).getOneBased();
+                if (Objects.equals(key, NAME_IDENTIFIER)) {
+                    tempStr += String.format("%s. %s.\n", index, personRelevantInfo.get(NAME_IDENTIFIER));
```

Comment:
I think here can use assert or j include test cases method in the ViewCOmmandTest.java file